### PR TITLE
Bug 1181238 - Navigating to a new URL while focused on a text field breaks browser layout

### PR DESCRIPTION
**Pull request by bkmunar**
1. adjustFooterSize is causing the constraint problems, especially when there's keyboardState and no snackbars. So I just checked if there really are snackbars in the UIView before carrying out the remaking of constraints, else, keep as if there were no snackbars.

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -976,7 +976,9 @@ extension BrowserViewController: BrowserDelegate {
             let bars = self.snackBars.subviews
             // if the keyboard is showing then ensure that the snackbars are positioned above it, otherwise position them above the toolbar/view bottom
             if let state = keyboardState {
-                make.bottom.equalTo(-(state.height + 20))
+                if bars.count > 0 {
+                    make.bottom.equalTo(-(state.height + 20))
+                }
```
Comment (thebnich) [inline Client/Frontend/Browser/BrowserViewController.swift]: Let's keep the `bottom` and `top` constraints together to help avoid similar mistakes in the future. I'd suggest grouping them like this:

```
if let state = keyboardState where bars.count > 0 {
    let view = bars[bars.count-1] as! UIView
    make.top.equalTo(view.snp_top)
    make.bottom.equalTo(-(state.height + 20))
} else {
    make.top.bottom.equalTo(self.toolbar?.snp_top ?? self.view.snp_bottom)
}
```

**Comment by fxios-buildasaur** [comment]
Result of integration 1
Integration took 9 minutes and 23 seconds.
All 0 tests passed, but please fix 21 warnings.

**Comment by thebnich** [comment]
Looks good to me, but please remove the `HEAD is now at 9f67f17` part of your commit message ;)

**Comment by bkmunar** [comment]
ugh @thebnich u always catch me

**Pull request merged**